app/src/scrcpy.c

```diff
@@ -80,7 +82,9 @@ static SDL_bool event_loop(void) {
                 if (!screen.has_frame) {
                     screen.has_frame = SDL_TRUE;
                     // this is the very first frame, show the window
-                    screen_show_window(&screen);
+                    if (!no_window) {
+                        screen_show_window(&screen);
+                    }
```

Comment:
I think you can skip code far earlier, you may not even need to initialize SDL nor decode frames at all if you have no window.